### DISCUSSION TITLE
Customising Orin NX boot order

### DISCUSSION THREAD

**Discussion by isaacporras** [OE4T/meta-tegra#1416] [Q&A] [answered]
Hi,

So currently I'm have to select the boot order manually for my custom Orin NX board using the UEFI menu to boot up correctly. The boot order now tries to load over HTTP IPv6. It shows this when trying to boot up:

```
Jetson UEFI firmware (version v35.3.1 built on 2023-01-24T13:18:32+00:00)
ESC   to enter Setup.
F11   to enter Boot Manager Menu.
Enter to continue boot.
**  WARNING: Test Key is used.  **
......
>>Start HTTP Boot over IPv6.
```

I want to know if there is a way to change the default boot order so I don't have to change it from UEFI menu for every board.
Thanks!

**Comment by joekale-pp** [comment, marked as answer]
You can add an overlay with the "DefaultBootPriority" UEFI Variable set and then append that overlay to the list of overlays to be applied on boot with the `TEGRA_BOOTCONTROL_OVERLAYS` variable in your machine conf. You can use the following as an example in a `dtso` file and then run `dtc -I dts -o dtb -o boot_priority.dtbo boot_priority.dtso`. I do this work in a bbappend file for `edk2-firmware-tegra` in my layer.

```
/dts-v1/;

/ {
        overlay-name = "Boot Priority Settings";

        fragment@0 {
                target-path = "/";

                board_config {
                        sw-modules = "uefi";
                };

                __overlay__ {
                        firmware {
                                uefi {
                                        variables {
                                                gNVIDIAPublicVariableGuid {
                                                        DefaultBootPriority {
                                                                data = "nvme, emmc, usb";
                                                                locked;
                                                        };
                                                };
                                        };
                                };
                        };
                };
        };
};
```

**Reply by isaacporras** [reply]
Thanks Joe, that worked!

**Comment by zachjacobs** [comment]
I'm seeing this too.  Any chance you can provide a bit more detail on how to do this?

**Reply by sarahn** [reply]
What details are you missing?

**Reply by dwalkes** [reply]
There's an example of applying a devicetree in the edk2-firmware-tegra recipe at https://github.com/OE4T/meta-tegra/commit/d421bf5cfe1c3ad876984d12a4a6c7158d67d6b7 if this is what you are looking for.  You could apply this in a bbappend in your layer instead.
